src/ophyd_async/tango/signal/_signal.py

```diff
@@ -0,0 +1,215 @@
+    timeout:
+        The timeout for the read and write operations
+    name:
+        The name of the Signal
+    """
+    if datatype is None:
+        datatype = infer_python_type(read_trl)
+    backend = make_backend(datatype, read_trl, write_trl or read_trl, device_proxy)
+    return SignalRW(backend, timeout=timeout, name=name)
+
+
+# --------------------------------------------------------------------
+def tango_signal_r(
+    datatype: Optional[Type[T]] = None,
+    *,
+    read_trl: str,
+    device_proxy: Optional[DeviceProxy] = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalR[T]:
+    """Create a `SignalR` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    datatype:
+        Check that the Attribute/Command is of this type
+    read_trl:
+        The Attribute/Command to read and monitor
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the read operation
+    name:
+        The name of the Signal
+    """
+    if datatype is None:
+        datatype = infer_python_type(read_trl)
+    backend = make_backend(datatype, read_trl, read_trl, device_proxy)
+    return SignalR(backend, timeout=timeout, name=name)
+
+
+# --------------------------------------------------------------------
+def tango_signal_w(
+    datatype: Optional[Type[T]] = None,
+    *,
+    write_trl: str,
+    device_proxy: Optional[DeviceProxy] = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalW[T]:
+    """Create a `TangoSignalW` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    datatype:
+        Check that the Attribute/Command is of this type
+    write_trl:
+        The Attribute/Command to write to
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the write operation
+    name:
+        The name of the Signal
+    """
+    if datatype is None:
+        datatype = infer_python_type(write_trl)
+    backend = make_backend(datatype, write_trl, write_trl, device_proxy)
+    return SignalW(backend, timeout=timeout, name=name)
+
+
+# --------------------------------------------------------------------
+def tango_signal_x(
+    write_trl: str,
+    device_proxy: Optional[DeviceProxy] = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalX:
+    """Create a `SignalX` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    write_trl:
+        The Attribute/Command to write its initial value to on execute
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the command operation
+    name:
+        The name of the Signal
+    """
+    backend = make_backend(None, write_trl, write_trl, device_proxy)
+    return SignalX(backend, timeout=timeout, name=name)
+
+
+# --------------------------------------------------------------------
+def tango_signal_auto(
```

Comment:
See my other comment about requiring type hints: I would prefer we kept that in ophyd-async. We should require the user to say "The control system should return this type" and not give them the option of saying "This signal is whatever the control system says it is".

Reply:
Annotated signals will now require type hints unless they are SignalX